android/modules/network/src/java/ti/modules/titanium/network/TiHTTPClient.java

```diff
@@ -1271,12 +1271,12 @@ protected void setUpClient(HttpURLConnection client, Boolean isPostOrPutOrPatch)
 		        return;
 		    }
 
-		    client.setUseCaches(true);
 		    client.setRequestMethod(method);
 		    client.setDoInput(true);
 
 		    if (isPostOrPutOrPatch) {
 		        client.setDoOutput(true);
+		        client.setChunkedStreamingMode(0);
```

Comment:
Some notes:
- This method is explicitly noted in the docs as to be used when we don't know the content length. But I think we always do know the content length (stored as totalLength in the connectionRunnable class?). When the length is known, they urge you to use #setFixedLengthStreamingMode(long)  https://developer.android.com/reference/java/net/HttpURLConnection.html#setFixedLengthStreamingMode(long)
- Secondly, when either of these are used, we avoid buffering which improves performance, but both have an important note in the docs:

> When output streaming is enabled, authentication and redirection cannot be handled automatically. A HttpRetryException will be thrown when reading the response if authentication or redirection are required. This exception can be queried for the details of the error.

As such, I think this change will actually break requests done against URLs that have either redirects or ask for basic auth. It'd be very useful to add some unit tests that cover these scenarios to verify.